massa-event-cache/src/event_cache.rs

```diff
@@ -0,0 +1,454 @@
+use std::cmp::max;
+// std
+use nom::AsBytes;
+use std::path::Path;
+// third-party
+use crate::ser_deser::{
+    SCOutputEventDeserializer, SCOutputEventDeserializerArgs, SCOutputEventSerializer,
+};
+use massa_models::execution::EventFilter;
+use massa_models::output_event::SCOutputEvent;
+use massa_serialization::{DeserializeError, Deserializer, Serializer};
+use rocksdb::{DBIterator, IteratorMode, WriteBatch, DB};
+use tracing::debug;
+
+const OPEN_ERROR: &str = "critical: rocksdb open operation failed";
+const CRUD_ERROR: &str = "critical: rocksdb crud operation failed";
+const EVENT_DESER_ERROR: &str = "critical: event deserialization failed";
+
+/// Module key formatting macro
+#[macro_export]
+macro_rules! event_key {
+    ($event_slot:expr) => {
+        [&$event_slot.to_bytes()[..], &[MODULE_IDENT]].concat()
+    };
+}
+
+pub(crate) struct EventCache {
+    /// RocksDB database
+    db: DB,
+    /// How many entries are in the db. Count is initialized at creation time by iterating
+    /// over all the entries in the db then it is maintained in memory
```

Comment:
The cache is not meant to be loaded from disk, we should clear it every time.
So this counter can safely start at 0